plugin/utils/gapic_utils.py

```diff
@@ -37,17 +37,39 @@ def read_from_gapic_yaml(yaml_file):
         gapic_yaml = yaml.load(f)
 
     collections = {}
+    fixed_collections = {}
+
+    all_collections = []
     if 'collections' in gapic_yaml:
-        collections = load_collection_configs(gapic_yaml['collections'],
-                                              collections)
+        all_collections = [gapic_yaml['collections']]
     for interface in gapic_yaml.get('interfaces', []):
         if 'collections' in interface:
-            collections = load_collection_configs(interface['collections'],
-                                                  collections)
-    fixed_collections = {}
+            all_collections.append(interface['collections'])
+
+    all_entities = []
+    for collection in all_collections:
+        all_entities.extend(collection)
+
+    (single_resource_names, fixed_resource_names) = \
+        find_single_and_fixed_entities(all_entities)
+
+    for entity in single_resource_names:
+        collections = load_collection_configs(entity, collections)
```

Comment:
You have modified load_collection_configs to operate on a single entity instead of a list of entities, but then used it in a loop. Can we just leave load_collection_configs unmodified?

Reply:
sure